platform/mds/mds/src/main/java/org/motechproject/mds/repository/MotechDataRepository.java

```diff
@@ -177,9 +184,9 @@ public long count(InstanceSecurityRestriction restriction) {
     }
 
     public long count(String[] properties, Object[] values, InstanceSecurityRestriction restriction) {
-        Query query = createQuery(properties, values, restriction);
-        QueryUtil.setCountResult(query);
-        return (long) QueryExecutor.executeWithArray(query, values, restriction);
+        // This method is a workaround for avoiding .setUnique in .retrieve
```

Comment:
use retrieveAll then. Although, how is count doing setUnique? Isn't this a widely used method?